src/DeviceInput/deviceInputSystem.ts

```diff
@@ -276,7 +282,10 @@ export class DeviceInputSystem implements IDisposable {
             if (kbKey) {
                 kbKey[evt.keyCode] = 1;
                 if (this.onInputChanged) {
-                    this.onInputChanged(DeviceType.Keyboard, 0, evt.keyCode, 0, kbKey[evt.keyCode]);
+                    const eventData: { [k: string]: any } = {};

```

Comment:
Doesn't this create a new object each time? Maybe the object could be created outside (at the file level, or once in the constructor) and reused here (and below)?

Reply:
native (browser) event data is usually immutable (the object itself), i think that in this case it is fine to not reuse the object to prevent from people using the vent data incorrectly.